tests/ecosystem/upgrade/test_logging_upgrade.py

```diff
@@ -0,0 +1,201 @@
+import logging
+import pytest
+
+from ocs_ci.ocs.cluster import CephCluster, CephHealthMonitor
+from ocs_ci.utility.deployment_openshift_logging import (
+    check_health_of_clusterlogging, get_clusterlogging_subscription
+)
+from ocs_ci.framework import config
+from ocs_ci.ocs.resources.pod import get_all_pods
+from ocs_ci.framework.pytest_customization.marks import (
+    post_ocp_upgrade
+)
+from ocs_ci.ocs import constants, ocp
+from ocs_ci.utility.utils import (
+    TimeoutSampler, get_ocp_version, run_cmd
+)
+
+logger = logging.getLogger(__name__)
+
+
+def check_cluster_logging():
+    """
+    Few checks to assert for before and after upgrade
+    1. Checks for pods in openshift-logging project
+    2. Checks for the health of logging cluster before upgrade
+
+    """
+
+    assert get_all_pods(
+        namespace=constants.OPENSHIFT_LOGGING_NAMESPACE
+    ), 'Some or all pods missing in namespace'
+
+    assert check_health_of_clusterlogging(), "Cluster is not Healthy"
+
+
+def check_csv_logging_upgrade_phase():
+    """
+    The function checks in the CSV for the phase "Succeeded"
+    for cluster-logging operator and Elastic-search
+    operator
+
+    Returns:
+        bool: True if CSV status is Succeeded, False otherwise
+
+    """
+
+    csv = ocp.OCP(
```

Comment:
Do you know we have CSV class directly in the resources?
https://github.com/red-hat-storage/ocs-ci/blob/045b0a5821a2ce6dc54956ddac532df8425bb8d3/ocs_ci/ocs/resources/csv.py

So please use directly CSV if you want to have CSV resource.
Also we have already implemented some functionality related to phase under OCP class here:

https://github.com/red-hat-storage/ocs-ci/blob/045b0a5821a2ce6dc54956ddac532df8425bb8d3/ocs_ci/ocs/ocp.py#L744